test/jdk/java/awt/Scrollbar/ScrollbarKeyControlTest.java

```diff
@@ -0,0 +1,165 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4943277
+  @requires (os.family == "linux")
+  @summary XAWT: Scrollbar can't be controlled by keyboard
+  @key headful
+*/
+
+import java.awt.AWTException;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.Scrollbar;
+import java.awt.Toolkit;
+
+import java.awt.event.AdjustmentListener;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+public class ScrollbarKeyControlTest implements AdjustmentListener, KeyListener {
+    Scrollbar scrollbarV;
+    Scrollbar scrollbarH;
+    volatile int changesTotal = 0;
+    Robot robot;
+    Object LOCK = new Object();
+    Frame frame;
+
+    public static void main(String[] args) throws Exception {
+        if (!System.getProperty("os.name").startsWith("Linux")) {
+            System.out.println("This test is for XAWT only.");
+            return;
+        }
+        ScrollbarKeyControlTest scrollbarKeyControlTest = new ScrollbarKeyControlTest();
+        scrollbarKeyControlTest.init();
+    }
+
+    public void init() throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                frame = new Frame("Scrollbar Test");
+
+                scrollbarV = new Scrollbar(Scrollbar.VERTICAL, 0, 1, 0, 255);
+                scrollbarH = new Scrollbar(Scrollbar.HORIZONTAL, 0, 60, 0, 300);
+
+                scrollbarH.addAdjustmentListener(this);
+                scrollbarH.addKeyListener(this);
+                scrollbarV.addAdjustmentListener(this);
+                scrollbarV.addKeyListener(this);
+
+                frame.add("South", scrollbarH);
+                frame.add("East", scrollbarV);
+
+                frame.setSize(200, 200);
+                frame.setVisible(true);
+                frame.validate();
+                frame.toFront();
+            });
+            robot = new Robot();
+            testOneScrollbar(scrollbarV);
```

Comment:
you should need to add robot.delay(1000) after frame is visible before commencing test

Reply:
Updated.